libs/q/CMakeLists.txt

```diff
@@ -1,16 +1,20 @@
-
 find_source_tree( LIBQ_HEADERS          "Header Files"          include/q "*.hpp" )
 find_source_tree( LIBQ_INTERNAL_HEADERS "Internal Header Files" src       "*.hpp" )
 find_source_tree( LIBQ_SOURCES          "Source Files"          src       "*.cpp" )
 
-add_library( q ${LIBQ_HEADERS} ${LIBQ_INTERNAL_HEADERS} ${LIBQ_SOURCES} )
+add_library( q ${LIBQ_SOURCES} )
```

Comment:
This removes the header files from the project, when working in an IDE (Visual Studio, XCode, etc).
I see in the article you linked to that this is the way it "should be done". There's _certainly_ something wrong with this approach.